src/greynirseq/utils/qa/lemmatizer.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+import argparse
+
+from reynir import Greynir, _Sentence
+from reynir.bintokenizer import TokenList
+from reynir.ifdtagger import IFD_Tagset
+
+from greynirseq.nicenlp.models.icebert import IcebertModel
+
+from typing import List, Tuple
+
+
+TokLem = Tuple[List[str], List[str]]
+
+
+class Lemmatizer:
+    g = None
+    ib = None
+    device = "cpu"
+
+    def __init__(self, use_icebert: bool = True) -> None:
+        self.g = Greynir()
+        if use_icebert:
+            self.ib = IcebertModel.pos_from_settings()
+
+    def _fallback(self, sentence: _Sentence) -> TokLem:
+        tokens = [t.txt for t in sentence.tokens]
+        return tokens, tokens
+
+    def lemmatize_sentence(self, sentence: _Sentence) -> TokLem:
+        if sentence.tree is not None:
+            return self.g_lemmatize(sentence)
+        else:
+            a_lemmas = []
+            a_tokens = []
+            tokens = sentence.tokens
+            # Split on 100 words to not hit 512 token limit in IceBERT
+            for i in range(0, len(tokens), 100):
+                p_lemmas, p_tokens = self.ib_lemmatize(tokens[i * 100 : (i + 1) * 100])
```

Comment:
Does this work well in practice? (As opposed to e.g. using window size 100 and overlap/stride of 50)

Reply:
On the rare occasion a sentence is this long it worked alright in practice, but could be made more robust if necessary. Leaving a note in comment